install/roles/certificates-all/tasks/main.yml

```diff
@@ -55,6 +55,35 @@
     delete: yes
     comment: Allow LetsEncrypt temporarily
 
+- name: Get the certificates list
+  register: certs
+  find:
+    path: '/etc/letsencrypt/archive/'
+    recurse: yes
+    patterns: 'cert*,chain*,fullchain*'
+
+- name: Get the keys list
+  register: keys
+  find:
+    path: '/etc/letsencrypt/archive/'
+    recurse: yes
+    patterns: 'privkey*'
+
+# Posix permissions: files should be readable, by default, only by root.
+# The ACL will extend this to other daemons
+- name: Set the unix mode for the certificate files readable by root only
```

Comment:
(Forgot to click "add comment"…)

It seems fine for a first time deployment. However, if running the playbook on an already deployed server, and when pebble has been restarted (or the certs need to be generated), slapd fails to restart at some point.

Setting the permissions to `600` resets the mask of the existing ACL (same as noted in cert-perms), and the effective permission for the user openldap is `user:openldap:r--		#effective:---`. Then later, the ldap role calls cert-perms and fixes the ACL and mask.

But, in that case, when the certificate is generated, the renewal-hook is run and restarts slapd. It does not happen on a first install since the renewal-hook is configured after the cert-perms role is applied.

So, some options that could be tried:
- find a permissions+ACL combination that doesn't imply a reset of the mask, and maybe gather everything in certificates-all ;
- add a workaround to make sure slapd is started after the cert-perms are applied, and before slapd is used (in the ldap role), with a `meta: flush_handlers`.

Glad to finally understand that one :sweat_smile:

Reply:
I found the issue in the, it is now fixed.
I will double check if this has been pushed or not.

Reply:
Yes, the fix is in the branch, commit 408b62a.
I am now working on the email copied in the Sent folder

Reply:
Yes, that's the change I am commenting on. The behavior I described is still happening with on this branch.
Anyway, this is separate from your work on postfix/dovecot, these changes could be kept for another PR.

Reply:
I did not read your description carefully enough, I will fix this before merging, then.

Reply:
This is now fixed, and does not reset the ACL any more. The chmod is only applied when the certificates are created for the first time, with the mode as 0644.